### PR TITLE
Fixed authorization & Playback failures fixed 

### PR DESCRIPTION
**Pull request by Computershik73**

### CODE REVIEW

**Review by neochapay — APPROVED** [review]

File: src/YaSailMusic.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include <QScopedPointer>
 #include <QtGlobal>
 
-#include <sailfishapp.h>
+#include </home/okabe2011/SailfishOS/mersdk/targets/SailfishOS-3.4.0.24-armv7hl.default/usr/include/sailfishapp/sailfishapp.h>
```
Comment (neochapay) [inline src/YaSailMusic.cpp:12]: Full patch

File: src/YaSailMusic.cpp [hunk 2]
```diff
@@ -32,12 +33,13 @@ int main(int argc, char *argv[])
     QScopedPointer<QGuiApplication> application(SailfishApp::application(argc, argv));
     application->setOrganizationName(QStringLiteral("org.ilyavysotsky"));
     application->setApplicationName(QStringLiteral("yasailmusic"));
+
```
Comment (neochapay) [inline src/YaSailMusic.cpp:36]: Empty line

File: src/apirequest.cpp [hunk 3]
```diff
@@ -23,17 +36,46 @@ void ApiRequest::makeApiGetRequest(const QString &method, const QUrlQuery &q) {
     url.setQuery(query.query());
     QNetworkRequest request(url);
     request.setRawHeader("Authorization", "OAuth "+accessToken.toUtf8());
-    connect(_manager, &QNetworkAccessManager::finished, this, &ApiRequest::dataReady, Qt::QueuedConnection);
+    request.setRawHeader("Accept", "*/*");
+    request.setRawHeader("X-Yandex-Music-Content-Type", "adult");
+    request.setHeader(QNetworkRequest::UserAgentHeader, "Maple/519 (iPhone; iOS 12.2; Scale/2.00)");
```
Comment (neochapay) [inline src/apirequest.cpp:41]: Really needed ?

File: src/apirequest.cpp [hunk 4]
```diff
@@ -42,4 +84,5 @@ void ApiRequest::dataReady(QNetworkReply *reply) {
         qDebug() << data;
 
     }
+
```
Comment (neochapay) [inline src/apirequest.cpp:87]: Empty line

File: src/authorization.h [hunk 5]
```diff
@@ -21,6 +21,8 @@ class Authorization : public QObject
     static void setupRequest(QNetworkRequest *r);
     Q_INVOKABLE void doAuth(QString username, QString password);
     Q_INVOKABLE bool checkToken();
+    Q_INVOKABLE void removeAccessToken();
+    Q_INVOKABLE void removeUserId();
```
Comment (neochapay) [inline src/authorization.h:25]: Why not use exit() function

File: src/models/playlistmodel.cpp [hunk 6]
```diff
@@ -172,15 +196,58 @@ void PlaylistModel::playTrack()
...
+    /* query.addQueryItem("type", type);
+    if (type.contains("trackStarted")) {
+        query.addQueryItem("totalPlayedSeconds", "0");
+    } else {
+        query.addQueryItem("totalPlayedSeconds", QString::number(m_playList.at(m_currentIndex)->duration));
+    }
+    query.addQueryItem("track-id", QString::number(m_playList.at(m_currentIndex)->trackId)+":"+QString::number(m_playList.at(m_currentIndex)->albumCoverId));
+    query.addQueryItem("timestamp", curdt);*/
+    QJsonObject o1;
+    if (type.contains("trackStarted")) {
```
Comment (neochapay) [inline src/models/playlistmodel.cpp:230]: Codestyle

File: src/models/playlistmodel.cpp [hunk 7]
```diff
@@ -197,6 +264,7 @@ void PlaylistModel::loadMyWave()
     }
     m_api->makeApiGetRequest("/rotor/station/user:onyourwave/tracks", query);
     connect(m_api, &ApiRequest::gotResponse, this, &PlaylistModel::getWaveFinished);
+
```
Comment (neochapay) [inline src/models/playlistmodel.cpp:267]: Empty line

File: src/models/playlistmodel.h [hunk 8]
```diff
@@ -59,6 +61,7 @@ private slots:
 private:
     bool m_loading;
     int m_currentIndex;
+    QString batchid;
```
Comment (neochapay) [inline src/models/playlistmodel.h:64]: Codestyle

**Review by neochapay — CHANGES_REQUESTED** [review]

File: src/apirequest.cpp (hunk 3 above)
Comment (Computershik73) [reply src/apirequest.cpp:41]: Yes, these headers are used in such requests in the official app.

**Pull request merged**